src/meili/map_matcher_factory.cc

```diff
@@ -74,9 +87,28 @@ MapMatcherFactory::MergeConfig(const std::string& name,
     }
   }
 
+  std::unordered_set<std::string> customizable;
+  for (const auto& item : config_.get_child("customizable")) {
```

Comment:
@ericrwolfe @kevinkreiser @dnesbitt61 does it make sense to put the "customizable" filtering logic somewhere else? It's actually like `service_limits` to avoid clients passing insane parameters (e.g. very large `search_radius`) to meili. Having this logic in the factory would surprise users who use meili as a library.

Reply:
I think this just says which config settings can be overriden in a request. There should be some bounds checking to make sure the inputs are sane.

Reply:
This checks that the user parameters passed in via `trace_options` have been allowed by the `meili.customizable` array in `valhalla.json`. Most of these parameters are bounds-checked elsewhere, e.g. [`search_radius` is checked against `max_search_radius`](https://github.com/valhalla/valhalla/blob/30768498d75fe3c5d0a171e3aab26dc0f8a5064f/src/meili/map_matcher.cc#L706-L707).

@ptpt do you mean that users who use meili as a service library would wouldn't define the `meili.customizable` property in `valhalla.json` and therefore the "custom" parameters they set wouldn't work as expected?

Reply:
Thanks @ptpt. I'll defer to @dnesbitt61 or @kevinkreiser on where something like that should live, and whether it makes sense as part of this PR or a follow on.